src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,57 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.Arrays;
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private static final int THRESHOLD = 4;
+    private final int[] array;
 
     public MergeSortAction(int[] array) {
+        this.array = array;
+    }
+
+    @Override
+    protected void compute() {
+        if (array.length <= THRESHOLD) {
+            sort(array);
+        } else {
+            int middle = array.length / 2;
+            int[] leftArray = Arrays.copyOfRange(array, 0, middle);
+            int[] rightArray = Arrays.copyOfRange(array, middle, array.length);
+
+            MergeSortAction leftAction = new MergeSortAction(leftArray);
+            MergeSortAction rightAction = new MergeSortAction(rightArray);
+            leftAction.fork().join();
+            rightAction.fork().join();
```

Comment:
don't join immediately after fork, here, there is no concurrency, we first wait for left action to complete, only then start right action